packages/flutter/analysis_options.yaml

```diff
@@ -5,3 +5,7 @@ include: ../analysis_options.yaml
 analyzer:
   enable-experiment:
     - non-nullable
+  errors:
+    always_require_non_null_named_parameters: false # not needed with nnbd
+    void_checks: false # issues with nnbd
+    unnecessary_null_comparison: false
```

Comment:
Why do we turn this one off?

Reply:
This is needed to avoid warning on null assertion on non-nullable variable. For instance:

```dart
void m(String s) {
  assert(s != null);
}
```

This could change with https://github.com/flutter/flutter/issues/61042

Reply:
If the variable itself is already non-nullable, why do we still need the assert? Wouldn't we want to clean that up and remove it?

Reply:
Either way, It would probably be good to have a comment on this line explaining why we turn this off.

Reply:
>If the variable itself is already non-nullable, why do we still need the assert?

See https://github.com/dart-lang/language/issues/1018:

> At first glance, it might seem sensible to remove the assertion on the grounds that c can no longer be null, but that's only true in strong mode. Flutter will have a long tail of users running in weak mode until Dart 3.0, and assertions like these are necessary to help prevent them from passing nulls to Flutter APIs.

Reply:
Thanks for the context and adding it as a comment.